operator/check_operator.go

```diff
@@ -0,0 +1,157 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/engine"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/policy"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+	preflighterr "github.com/redhat-openshift-ecosystem/openshift-preflight/errors"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/lib"
+)
+
+type Option = func(*operatorCheck)
+
+// TODO(): replace this value when the default in package cmd is moved to a central location
+const defaultScorecardWaitTime = "240"
+
+// NewCheck is a check runner that executes the Operator Policy.
+func NewCheck(image, kubeconfig, indeximage string, opts ...Option) *operatorCheck {
```

Comment:
[Feedback Request]: This signature contains the kubeconfig and is expecting a file. Ideally, we would want to accept an io.Reader but there are other changes that need to happen here before we do that. I'm taking suggestions about what to do here so we don't have to change this signature once we do make the changes necessary to use an io.Reader and cause potential breakages when consumers upgrade versions of this library.

I could accept the Kubeconfig as an option instead, and differentiate similarly to how I'm currently doing the DockerConfigJSON, but a Kubeconfig isn't optional.

Reply:
Let's iterate on this in the branch.